### PR TITLE
Squeeze the V2 schema and allow building V2 SDKs

### PR DESCRIPTION
**Pull request by thomas11**
- Upgrade schema-tools to 0.2.2 for squeeze improvement
- First iteration of squeezing the v2 spec
- Rename files and code for clarity, remove some unused code
- Move squeezing to just before schema generation
- Preserve resources in v1-lock when squeezing
- Groundwork for generating the v2 schema, but not plumbed all the way through yet.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,8 @@ GOEXE ?= $(shell go env GOEXE)
 LOCAL_PROVIDER_FILENAME := $(PROVIDER)$(GOEXE)
 LOCAL_ARM2PULUMI_FILENAME := arm2pulumi$(GOEXE)
 
+VERSION_DIR ?= 
+
```
Comment (danielrbradley) [inline Makefile:28]: I think this is the only part of this PR I'm not sure about yet. It feels like it's adding a bit of complexity - but is only needed for a temporary transition period. 

Do we need the ability to generate the SDKs on the same branch as v1?

Could we just do the v2 prerelease from a branch which only contains a minimal change to switch which schema we use? Then can rebase/re-merge the v2 prerelease branch as we continue making changes on master?

**Review by danielrbradley — APPROVED** [review]
Approving for the schema work to go in. We'll pull out the 2 commits for the SDK building into a separate follow-on PR.

File: provider/pkg/versioning/gen.go [hunk 2]
```diff
@@ -98,12 +133,17 @@ func ReadVersionSources(rootDir string) (VersionSources, error) {
 		return VersionSources{}, err
 	}
 
-	v1Spec, err := ReadDefaultConfig(path.Join(rootDir, "versions", "v1-spec.yaml"))
+	knownExplicitResources, err := ReadRequiredExplicitResources(path.Join(rootDir, "versions", "required-explicit-resources.txt"))
```
Comment (thomas11) [inline provider/pkg/versioning/gen.go:136]: Where does this file come from @danielrbradley?

Comment (thomas11) [reply provider/pkg/versioning/gen.go:136]: Ah, Metabase. Not too obvious to a reader not familiar with this area.

**Pull request merged**